pkg/hakeeper/rsm.go

```diff
@@ -584,14 +584,20 @@ func (s *stateMachine) Lookup(query interface{}) (interface{}, error) {
 
 func (s *stateMachine) SaveSnapshot(w io.Writer,
 	_ sm.ISnapshotFileCollection, _ <-chan struct{}) error {
-	// FIXME: ready to use gogoproto to marshal the state, just need to figure
-	// out how to write to the writer.
-	enc := gob.NewEncoder(w)
-	return enc.Encode(s.state)
+	data := make([]byte, s.state.Size())
```

Comment:
snapshot creation is currently designed to be very frequent, and mpool should be used here to avoid frequent make memory

Reply:
This PR mainly replace `encoding/gob`, I suggest that leaving further optimization to related co-workers and 
 optimizing when this would be a bottleneck.